### PR TITLE
[M2-2961]Feature: activity flow passing business logic

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/shared/utils/hooks/useEntityType.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  flowId: string
+}
+
+type RegularReturn = {
+  isFlow: false
+}
+
+type Return = FlowReturn | RegularReturn
+
+export const useFlowType = (): Return => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/shared/utils/hooks/useEntityType.ts:14]: Done

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const completeEntity = () => {
+    clearActivityItemsProgressById(props.activityId, props.eventId)
+    entityCompleted({
+      appletId: props.appletDetails.id,
+      entityId: props.flowId ? props.flowId : props.activityId,
+      eventId: props.eventId,
+    })
+  }
+
+  const flowCompleted = (flowId: string) => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:29]: Done

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    const { activityFlows } = props.appletDetails
+
+    const currentFlow = activityFlows.find(flow => flow.id === flowId)!
+
+    const currentActivityIndexInFlow = currentFlow.activityIds.findIndex(activityId => activityId === props.activityId)!
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:34]: Fixed it

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        entityId: props.flowId ? props.flowId : props.activityId,
+        eventId: props.eventId,
+      })
+
+      if (props.publicAppletKey) {
+        return navigator.navigate(ROUTES.thanks.navigateTo(props.publicAppletKey, true))
+      }
+
+      return navigator.navigate(ROUTES.thanks.navigateTo(props.appletDetails.id, false))
+    }
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:60]: Done

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    return navigator.navigate(
+      ROUTES.activityDetails.navigateTo({
+        appletId: props.appletDetails.id,
+        activityId: nextActivityId,
+        eventId: props.eventId,
+        entityType: "flow",
+        flowId: props.flowId,
+      }),
+    )
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:83]: Done

[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:83]: Done

File: src/widgets/ActivityDetails/ui/AssessmentLayoutHeader.tsx [hunk 6]
```diff
@@ -21,14 +24,16 @@ export const AssessmentLayoutHeader = (props: Props) => {
...
 
-  const onSaveAndExitClick = () => {
-    return navigator.goBack()
-  }
-
   const cutStringToLength = (str: string, length: number) => {
     return str.length > length ? `${str.substring(0, length)}...` : str
   }
 
+  const onClick = () => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/ui/AssessmentLayoutHeader.tsx:31]: Done

File: src/widgets/ActivityDetails/model/hooks/useAnswers.ts [hunk 7]
```diff
@@ -36,6 +36,12 @@ export const useAnswer = (props: UseAnswerProps) => {
 
   const { getGroupInProgressByIds } = activityModel.hooks.useActivityGroupsInProgressState()
 
+  const getSubmitId = useCallback((groupInProgress: activityModel.types.ProgressState): string => {
+    const isFlow = groupInProgress.type === ActivityPipelineType.Flow
+
+    return isFlow ? groupInProgress.executionGroupKey : uuidV4()
+  }, [])
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/model/hooks/useAnswers.ts:43]: Please unwrap the function from userCallback

**Review by BamMironov — APPROVED** [review]

**Pull request merged**